homeassistant/components/solaredge_local/manifest.json

```diff
@@ -5,5 +5,9 @@
   "documentation": "https://www.home-assistant.io/integrations/solaredge_local",
   "iot_class": "local_polling",
   "loggers": ["solaredge_local"],
-  "requirements": ["solaredge-local==0.2.0"]
+  "requirements": [
+    "solaredge-local==0.2.1",
+    "uplink-protobuf==0.1.0",
+    "uplink==0.9.7"
```

Comment:
It doesn't look like these are used directly. You can min them to a minimum version in `solaredge-local` instead ie `uplink-protobuf>=0.1.0` , `uplink>=0.9.7`

Note that we don't generally accept libs that pin to a specific version, and you should use the minimum version needed.

Reply:
I tried that first! but the script have me error
```
$ python3 -m script.gen_requirements_all
******* ERROR
Errors while importing:  homeassistant.components.solaredge_local[Please pin requirement uplink-protobuf>=0.1.0, see https://developers.home-assistant.io/docs/creating_platform_code_review.html#1-requirements], homeassistant.components.solaredge_local[Please pin requirement uplink>=0.9.7, see https://developers.home-assistant.io/docs/creating_platform_code_review.html#1-requirements]
```

Reply:
If you release a new `solaredge-local` with those requirements and bump it here, that should go away

Reply:
```
% pip3 install --upgrade solaredge-local
Collecting solaredge-local
  Downloading solaredge_local-0.2.2.tar.gz (12 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/v7/0sj93k_n0p56bpyk3vmf2z900000gn/T/pip-install-_1_cmv__/solaredge-local_d8ae0416447c4acd98d93a218dd000e4/setup.py", line 8, in <module>
          import solaredge_local
        File "/private/var/folders/v7/0sj93k_n0p56bpyk3vmf2z900000gn/T/pip-install-_1_cmv__/solaredge-local_d8ae0416447c4acd98d93a218dd000e4/solaredge_local/__init__.py", line 1, in <module>
          from solaredge_local.solaredge import SolarEdge
        File "/private/var/folders/v7/0sj93k_n0p56bpyk3vmf2z900000gn/T/pip-install-_1_cmv__/solaredge-local_d8ae0416447c4acd98d93a218dd000e4/solaredge_local/solaredge.py", line 1, in <module>
          from uplink import Consumer, get
      ModuleNotFoundError: No module named 'uplink'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.0.1 -> 23.1
[notice] To update, run: pip install --upgrade pip
```

It looks like the requirements are not setup correctly in the package

Reply:
The package is importing the lib before its installed

```diff
diff --git a/setup.py b/setup.py
index 8851103..e21e929 100644
--- a/setup.py
+++ b/setup.py
@@ -1,11 +1,7 @@
 from __future__ import print_function
-from setuptools import setup, find_packages
+from setuptools import setup
 import io
-import codecs
 import os
-import sys
-
-import solaredge_local
 
 here = os.path.abspath(os.path.dirname(__file__))
 
```

That should fix it

Reply:
Yep, same error I got.. that's what confused me. Between the last working version 0.2.0 and this, only the versions are bumped. File changes are auto-generated classes. Not much in the setup at all.

Reply:
> But you'll need another release of the package

Thanks a lot. I'll make a pr and get this going ☺️